bucketstracking_agentmanager.go

```diff
@@ -110,13 +109,15 @@ func CreateBucketsTrackingAgentManager(ctx context.Context, opts BucketsTracking
 			tlsConfig:     opts.TLSConfig,
 			authenticator: opts.Authenticator,
 			httpTransport: httpTransport,
-			latestConfig:  bootstrapConfig,
 		},
 	}
+	m.state.latestConfig.Store(bootstrapConfig)
+
+	mgmtEndpoints := m.mgmtEndpointsLocked(bootstrapConfig)
```

Comment:
So when we call this when we are starting CNG  `srcHTTPAddrs` will just be what we pass using `--cb-host`? Whereas `mgmtEndpoints` will be the addresses of all the nodes on the cluster gotten in the bootstrap? What's the practical implications of using `mgmtEndpoints` to create the `NewBucketsWatcherHttp` instead of  `srcHTTPAddrs`?

Reply:
Possibly. Ignoring CNG srcHostAddrs is just the set of http addresses that the user provides (so maybe for CNG), mgmtEndpoints is all management endpoints on the cluster, which means that the watcher has access to use all them rather than a subset.

Reply:
I see, also why is it called `mgmtEndpointsLocked`?

Reply:
It's only safe to be called from inside the lock (once the manager has been returned from New)